src/derive_utils.rs

```diff
@@ -10,106 +10,242 @@ use crate::exceptions::PyTypeError;
+        for (out, arg) in output[..args_provided].iter_mut().zip(args) {
+            *out = Some(arg);
+        }
+
+        // Handle keyword arguments
+        let varkeywords = match (kwargs, self.accept_varkeywords) {
+            (Some(kwargs), true) => {
+                let mut varkeywords = None;
+                self.extract_keyword_arguments(kwargs, output, |name, value| {
+                    varkeywords
+                        .get_or_insert_with(|| PyDict::new(kwargs.py()))
+                        .set_item(name, value)
+                })?;
+                varkeywords
+            }
+            (Some(kwargs), false) => {
+                self.extract_keyword_arguments(kwargs, output, |name, _| {
+                    Err(self.unexpected_keyword_argument(name))
+                })?;
+                None
+            }
+            (None, _) => None,
+        };
+
+        // Check that there's sufficient positional arguments once keyword arguments are specified
+        if args_provided < self.required_positional_parameters {
+            let missing_positional_arguments: Vec<_> = self
+                .positional_parameter_names
+                .iter()
+                .take(self.required_positional_parameters)
+                .zip(output.iter())
+                .filter_map(|(param, out)| if out.is_none() { Some(*param) } else { None })
+                .collect();
+            if !missing_positional_arguments.is_empty() {
+                return Err(
+                    self.missing_required_arguments("positional", &missing_positional_arguments)
+                );
+            }
+        }
+
+        // Check no missing required keyword arguments
+        let missing_keyword_only_arguments: Vec<_> = self
+            .keyword_only_parameters
+            .iter()
+            .zip(&output[num_positional_parameters..])
+            .filter_map(|(keyword_desc, out)| {
+                if keyword_desc.required && out.is_none() {
+                    Some(keyword_desc.name)
+                } else {
                     None
-                } else if i < nargs {
-                    used_args += 1;
-                    Some(args.get_item(i))
+                }
+            })
+            .collect();
+
+        if !missing_keyword_only_arguments.is_empty() {
+            return Err(self.missing_required_arguments("keyword", &missing_keyword_only_arguments));
+        }
+
+        Ok((varargs, varkeywords))
+    }
+
+    #[inline]
+    fn extract_keyword_arguments<'p>(
+        &self,
+        kwargs: &'p PyDict,
+        output: &mut [Option<&'p PyAny>],
+        mut unexpected_keyword_handler: impl FnMut(&'p PyAny, &'p PyAny) -> PyResult<()>,
+    ) -> PyResult<()> {
+        let (args_output, kwargs_output) =
+            output.split_at_mut(self.positional_parameter_names.len());
+        let mut positional_only_keyword_arguments = Vec::new();
+        'kwarg_loop: for (kwarg_name, value) in kwargs {
```

Comment:
I think this label is unnecessary.

Reply:
Good catch; it was needed before the other changes to `position` / `find`!